packages/core/src/typeschema/types.ts

```diff
@@ -138,7 +146,26 @@ export function loadDataType(sd: StructureDefinition, profileUrl?: string | unde
 
   const dataTypes = getDataTypesMap(profileUrl);
 
-  dataTypes[sd.name] = schema;
+  // By default, only index by "type" for "official" FHIR types
+  // Note that if profileUrl is provided, then dataTypes will be PROFILE_DATA_TYPES[profileUrl].
+  if (
+    sd.url === `http://hl7.org/fhir/StructureDefinition/${sd.type}` ||
+    sd.url === `https://medplum.com/fhir/StructureDefinition/${sd.type}` ||
+    sd.type?.startsWith('http://') ||
+    sd.type?.startsWith('https://') ||
+    profileUrl
+  ) {
+    dataTypes[sd.type] = schema;
+  }
+
+  // Special cases by "name"
+  // These are StructureDefinitions that are technically "profiles", but are used as base types
+  if (!profileUrl) {
```

Comment:
It seems like it would be possible for a StructureDefinition to never wind up being added to `dataTypes` and just silently pass through `loadDataType`, i.e. if neither this if statement or the one before it passes. Compared to before where all SDs would definitely wind up being added via `dataTypes[sd.name] = schema`.

Should we be throwing if that happens?

Reply:
Summarizing offline discussion and changes:

It turns out the `profileUrl` param is 100% of the time the same as `StructureDefinition.url`.  I removed the param, and went through all of the cleanup.

Reply:
Fixed ✅